src/main/java/de/tum/in/www1/artemis/domain/Submission.java

```diff
@@ -110,12 +112,54 @@ public Result getLatestResult() {
         return null;
     }
 
+    /**
+     * Used to get result by correction round
+     * @param correctionRound to get result by
+     * @return the result based on the given correction-round
+     */
+    @Nullable
+    @JsonIgnore
+    public Result getResultByCorrectionRound(Long correctionRound) {
+        if (results != null && results.size() > correctionRound) {
+            return results.get(correctionRound.intValue());
+        }
+        return null;
+    }
+
+    /**
+     * Used to get result by correction round ignoring all automatic results
+     * @param correctionRound to get result by
+     * @return the result based on the given correction-round
+     */
+    @Nullable
+    @JsonIgnore
+    public Result getResultByCorrectionRoundIgnoreAutomatic(Long correctionRound) {
+        List<Result> withoutAutomaticResults = results.stream().filter(result -> result != null && !result.getAssessmentType().equals(AssessmentType.AUTOMATIC))
+                .collect(Collectors.toList());
+        if (withoutAutomaticResults != null && withoutAutomaticResults.size() > correctionRound) {
```

Comment:
I don't think `collect` returns `null`, but rather an empty collection, so we can drop that part of the check?

Reply:
yes you are right, I removed that check